dbt/adapters/sqlserver/connections.py

```diff
@@ -159,55 +236,53 @@ def open(cls, connection):
                 elif type_auth == "ActiveDirectoryMsi":
                     raise ValueError("ActiveDirectoryMsi is not supported yet")
 
-            elif type_auth == "ServicePrincipal":
-                app_id = getattr(credentials, "AppId", None)
-                app_secret = getattr(credentials, "AppSecret", None)
-
             elif getattr(credentials, "windows_login", False):
-                con_str.append(f"trusted_connection=yes")
+                con_str.append("trusted_connection=yes")
             elif type_auth == "sql":
-                #con_str.append("Authentication=SqlPassword")
+                con_str.append("Authentication=SqlPassword")
                 con_str.append(f"UID={{{credentials.UID}}}")
                 con_str.append(f"PWD={{{credentials.PWD}}}")
 
-            # still confused whether to use "Yes", "yes", "True", or "true"
```

Comment:
And here

Reply:
generally, I think it would be better if this only a PR that changes  `dbt-sqlserver`'s  `connections.py` to add the az cli options, rather than also introducing syntax from `dbt-synapse` especially because `dbt-synapse`'s `connections.py` will be going away soon anyway.

Reply:
agreed